subsys/bluetooth/controller/ll_sw/ull_sync.c

```diff
@@ -235,33 +235,20 @@ uint8_t ll_sync_terminate(uint16_t handle)
 {
 	memq_link_t *link_sync_lost;
 	struct ll_sync_set *sync;
-	uint32_t volatile ret_cb;
 	uint32_t ret;
-	void *mark;
 
 	sync = is_enabled_get(handle);
 	if (!sync) {
 		return BT_HCI_ERR_UNKNOWN_ADV_IDENTIFIER;
 	}
 
-	mark = ull_disable_mark(sync);
-	LL_ASSERT(mark == sync);
+	ret = ull_ticker_stop_with_mark(TICKER_ID_SCAN_SYNC_BASE + handle,
+					sync, NULL);
 
-	ret_cb = TICKER_STATUS_BUSY;
-	ret = ticker_stop(TICKER_INSTANCE_ID_CTLR, TICKER_USER_ID_THREAD,
-			  TICKER_ID_SCAN_SYNC_BASE + handle,
-			  ull_ticker_status_give, (void *)&ret_cb);
-	ret = ull_ticker_status_take(ret, &ret_cb);
 	if (ret) {
-		mark = ull_disable_mark(sync);
-		LL_ASSERT(mark == sync);
-
 		return BT_HCI_ERR_CMD_DISALLOWED;
 	}
 
```

Comment:
ull_disable call is needed. Rationale, event may be in pipeline, and ull_disable will flush it.

Reply:
`ull_ticker_stop_with_mark` will call `ull_disable` if no error occurred, as the current code also did.

Reply:
in the ull_sync case, NULL is passed

Reply:
@cvinayak Updated